libr/search/regexp.c

```diff
@@ -1,8 +1,87 @@
 /* radare - LGPL - Copyright 2008-2020 - pancake, TheLemonMan */
 
 #include "r_search.h"
+#include "search.h"
 #include <r_regex.h>
 
+R_IPI int search_regex_read(RSearch *s, ut64 from, ut64 to) {
+	RSearchKeyword *kw;
+	RListIter *iter;
+	RRegexMatch match;
+	RRegex rx = {0};
+	const int old_nhits = s->nhits;
+	int ret = 0;
+
+	r_list_foreach (s->kws, iter, kw) {
+		ut64 addr = from;
+		int reflags = R_REGEX_EXTENDED;
+
+		if (kw->icase) {
+			reflags |= R_REGEX_ICASE;
+		}
+
+		if (r_regex_init (&rx, (char *)kw->bin_keyword, reflags)) {
+			eprintf ("Cannot compile '%s' regexp\n", kw->bin_keyword);
+			return -1;
+		}
+
+		while (addr < to) { // get buffer
+			ut8 buf[512];
```

Comment:
handle r_cons_is_breaked() in this inner loop and store this buffer in the heap at the begining of the function, probably better to read blocks of 4KB or honor core->blocksize